cylc/flow/simulation.py

```diff
@@ -0,0 +1,219 @@
+        )
+
+
+def get_simulated_run_len(rtc: Dict[str, Any]) -> int:
+    """Get simulated run time.
+
+    rtc = run time config
+    """
+    limit = rtc['execution time limit']
+    speedup = rtc['simulation']['speedup factor']
+    if limit and speedup:
+        sleep_sec = (DurationParser().parse(
+            str(limit)).get_seconds() / speedup)
+    else:
+        sleep_sec = DurationParser().parse(
+            str(rtc['simulation']['default run length'])
+        ).get_seconds()
+
+    return sleep_sec
+
+
+def build_dummy_script(rtc: Dict[str, Any], sleep_sec: int) -> str:
+    """Create fake scripting for dummy mode.
+
+    This is for Dummy mode only.
+    """
+    script = "sleep %d" % sleep_sec
+    # Dummy message outputs.
+    for msg in rtc['outputs'].values():
+        script += "\ncylc message '%s'" % msg
+    if rtc['simulation']['fail try 1 only']:
+        arg1 = "true"
+    else:
+        arg1 = "false"
+    arg2 = " ".join(rtc['simulation']['fail cycle points'])
+    script += "\ncylc__job__dummy_result %s %s || exit 1" % (arg1, arg2)
+    return script
+
+
+def disable_platforms(
+    rtc: Dict[str, Any]
+) -> None:
+    """Force platform = localhost
+
+    Remove legacy sections [job] and [remote], which would conflict
+    with setting platforms.
+
+    This can be simplified when support for the FORBIDDEN_WITH_PLATFORM
+    configurations is dropped.
+    """
+    for section, keys in FORBIDDEN_WITH_PLATFORM.items():
+        if section in rtc:
+            for key in keys:
+                if key in rtc[section]:
+                    rtc[section][key] = None
+    rtc['platform'] = 'localhost'
+
+
+def parse_fail_cycle_points(
+    f_pts_orig: list
+) -> Union[None, List[PointBase]]:
+    """Parse `[simulation][fail cycle points]`.
+
+    - None for "fail all points".
+    - Else a list of cycle point objects.
+
+    Examples:
+        >>> this = parse_fail_cycle_points
+        >>> this(['all']) is None
+        True
+        >>> this([])
+        []
+    """
+    f_pts: Optional[List[PointBase]]
+    if 'all' in f_pts_orig:
+        f_pts = None
+    else:
+        f_pts = []
+        for point_str in f_pts_orig:
+            f_pts.append(get_point(point_str).standardise())
+    return f_pts
+
+
+def sim_time_check(
+    message_queue: 'Queue[TaskMsg]', itasks: List[TaskProxy]
+) -> bool:
+    """Check if sim tasks have been "running" for as long as required.
+
+    If they have change the task state.
+
+    Returns:
+        True if _any_ simulated task state has changed.
+    """
+    sim_task_state_changed = False
+    now = time()
+    for itask in itasks:
+        if itask.state.status != TASK_STATUS_RUNNING:
+            continue
+        # Started time is not set on restart
+        if itask.summary['started_time'] is None:
+            itask.summary['started_time'] = now
+        timeout = (
+            itask.summary['started_time'] +
+            itask.tdef.rtconfig['simulation']['simulated run length']
```

Comment:
Not a real config item, added to the `rtconfig['simulation']` dictionary at line 54.